packages/apollo-codegen-scala/src/codeGeneration.ts

```diff
@@ -157,9 +159,7 @@ export function classDeclarationForOperation(
             undefined,
             true
           );
-          const isOptional = !(
-            type instanceof GraphQLNonNull || type instanceof GraphQLNonNull
```

Comment:
Lol what even was this?

Reply:
@shadaj this is way back in history but can you recall if this was intentional or possibly a mistake?
Re: https://github.com/apollographql/apollo-tooling/commit/58f07c42be6f817b646cef002a366327683e0028#diff-508d6e2cd0e11941b1548a2ea48a909dR139

Reply:
Looks like a mistake! I really hope there's no situation in which double-checking a type is actually needed 😄

Reply:
@shadaj Was removing the second check the mistake, or was that correct? Like this was originally

`const isOptional = !(type instanceof GraphQLNonNull || type.ofType instanceof GraphQLNonNull);`

was the `ofType` incorrectly removed, or was that fine to remove?

Reply:
It was fine to remove the extra check, I believe that `ofType` was eliminated with changes to the IR format so only the top level type needed to be checked.